packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/MemberRemoverVisitor.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.SourceInput;
+
+namespace Microsoft.Generator.CSharp
+{
+    internal class MemberRemoverVisitor : LibraryVisitor
+    {
+        protected override MethodProvider? Visit(MethodProvider methodProvider)
+        {
+            foreach (var attribute in GetMemberSuppressionAttributes(methodProvider.EnclosingType))
+            {
+                if (ShouldRemove(methodProvider.EnclosingType, methodProvider.Signature, attribute))
+                {
+                    return null;
+                }
+            }
+
+            return methodProvider;
+        }
+
+        protected override ConstructorProvider? Visit(ConstructorProvider constructorProvider)
+        {
+            foreach (var attribute in GetMemberSuppressionAttributes(constructorProvider.EnclosingType))
+            {
+                if (ShouldRemove(constructorProvider.EnclosingType, constructorProvider.Signature, attribute))
+                {
+                    return null;
+                }
+            }
+
+            return constructorProvider;
+        }
+
+        protected override PropertyProvider? Visit(PropertyProvider propertyProvider)
```

Comment:
Were fields purposely left out ?

Reply:
AFAIK there is no support for customizing fields in autorest as they are typically not public.

Reply:
Sounds good.